TheMovieDb/Views/HomeViewController/HomeViewController.swift

```diff
@@ -0,0 +1,195 @@
+          HeaderCollectionView.self,
+          forSupplementaryViewOfKind: "HeaderCollectionView.reuseIdentifier",
+          withReuseIdentifier: HeaderCollectionView.reuseIdentifier)
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(refresh), for: .valueChanged)
+        collectionView.refreshControl = refreshControl
+        collectionView.sendSubviewToBack(refreshControl)
+        movieHomeCollectionView = collectionView
+    }
+    
+    @objc func refresh() {
+        presenter?.fetchAllMovieList()
+    }
+    
+    func showMoviesHome(arrMovie: [SectionMovie: [Movie]]) {
+        self.arrMovies = arrMovie
+        setUpCollectionView()
+        configureDataSource()
+        movieHomeCollectionView.reloadData()
+    }
+    
+    func showLoading() {
+        print("")
+    }
+    
+    func stopLoading() {
+       
+    }
+    
+    func configureDataSource() {
+        dataSource = UICollectionViewDiffableDataSource
+        <SectionMovie, Movie>(collectionView: movieHomeCollectionView) {
+            (collectionView: UICollectionView, indexPath: IndexPath, movieItem: Movie) -> UICollectionViewCell? in
+            guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: LargeCollectionViewCell.reuseIdentifer,
+                for: indexPath) as? LargeCollectionViewCell else { fatalError("Could not create new cell") }
+            cell.title = movieItem.title
+            cell.portraitPhotoURL = movieItem.posterPath
+            return cell
+        }
+        
+        dataSource.supplementaryViewProvider = { (
+          collectionView: UICollectionView,
+          kind: String,
+          indexPath: IndexPath) -> UICollectionReusableView? in
+
+          guard let supplementaryView = collectionView.dequeueReusableSupplementaryView(
+            ofKind: kind,
+            withReuseIdentifier: HeaderCollectionView.reuseIdentifier,
+            for: indexPath) as? HeaderCollectionView else { fatalError("Cannot create header view") }
+
+          supplementaryView.label.text = SectionMovie.allCases[indexPath.section].rawValue
+          return supplementaryView
+        }
+        let snapshot = snapshotForCurrentState()
+        dataSource.apply(snapshot, animatingDifferences: false)
+        
+    }
+    
+    func snapshotForCurrentState() -> NSDiffableDataSourceSnapshot<SectionMovie, Movie> {
+        var snapshot = NSDiffableDataSourceSnapshot<SectionMovie, Movie>()
+        for (key, value) in arrMovies {
+            snapshot.appendSections([key])
+            snapshot.appendItems(value)
+        }
+        return snapshot
+    }
+    
+}
+
+
+extension HomeViewController: SearchBarDelegate {
+    func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+       
+    }
+    
+    func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+       
+    }
+    
+    func updateSearchResults(for text: String) {
+        //
+    }
+    
+}
```

Comment:
If this is meant to be updated in the future please try to add `// TODO:` comments in the method's body.